tests/test_configuration.py

```diff
@@ -29,14 +29,17 @@ def test_returns_default_when_no_value_set(self) -> None:
         assert cfg.value == 42
 
     def test_raises_missing_configuration_when_value_is_none(self) -> None:
-        cfg = EnvValue("test")
+        cfg = EnvValue[str]("test")
 
         with pytest.raises(MissingConfigValue, match="'test' was not defined."):
             cfg.value  # pylint: disable=pointless-statement
 
     @patch.object(EnvValue, "getter", MagicMock(return_value=1))
```

Comment:
To make it more correct you should patch environ instead:
```suggestion
    @patch.dict(environ, {"RANDOM_VAR": "1"})
```